src/Symfony/Component/Message/README.md

```diff
@@ -0,0 +1,263 @@
+Message Component
+=================
+
+The Message component helps application to send and receive messages to/from other applications or via
+message queues.
+
+Resources
+---------
+
+  * [Contributing](https://symfony.com/doc/current/contributing/index.html)
+  * [Report issues](https://github.com/symfony/symfony/issues) and
+    [send Pull Requests](https://github.com/symfony/symfony/pulls)
+    in the [main Symfony repository](https://github.com/symfony/symfony)
+
+
+Documentation
+-------------
+
+**Note:** this documentation is to be moved to symfony.com when merging the Component.
+
+### Concepts
+
+![Component overview](Resources/doc/component-overview.png)
+
+1. **Sender**
+   Responsible for serializing and sending the message to _something_. This something can be a message broker or a 3rd 
+   party API for example.
+
+2. **Receiver**
+   Responsible for deserializing and forwarding the messages to handler(s). This can be a message queue puller or an API
+   endpoint for example.
+   
+3. **Handler**
+   Given a received message, contains the user business logic related to the message. In practice, that is just a PHP
+   callable.
+   
+
+### Bus
+
+The bus is used to dispatch messages. MessageBus' behaviour is in its ordered middleware stack. When using
+the message bus with Symfony's FrameworkBundle, the following middlewares are configured for you:
+
+1. `LoggingMiddleware` (log the processing of your messages)
+2. `SendMessageMiddleware` (enable asynchronous processing)
+3. `HandleMessageMiddleware` (call the registered handle)
+
+```php
+use App\Message\MyMessage;
+
+$result = $this->get('message_bus')->handle(new MyMessage(/* ... */));
+```
+
+### Handlers
+
+Once dispatched to the bus, messages will be handled by a "message handler". A message handler is a PHP callable 
+(i.e. a function or an instance of a class) that will do the required processing for your message. It _might_ return a
+result.
+
+```php
+namespace App\MessageHandler;
+
+use App\Message\MyMessage;
+
+class MyMessageHandler
+{
+    public function __invoke(MyMessage $message)
+    {
+        // Message processing...
+    }
+}
+```
+
+```xml
+<service id="App\Handler\MyMessageHandler">
+    <tag name="message_handler" />
+</service>
+```
+
+**Note:** If the message cannot be guessed from the handler's type-hint, use the `handles` attribute on the tag.
+
+### Asynchronous messages
+
+Using the Message Component is useful to decouple your application but it also very useful when you want to do some
+asychronous processing. This means that your application will produce a message to a queuing system and consume this
+message later in the background, using a _worker_.
+
+#### Adapters
```

Comment:
This component is kinda useless if we don't ship some adapters in core. At least AMQP and major cloud providers should be supported IMO.

Reply:
I wouldn't say useless as it serves the use cases of CQRS-type of architecture, which is not negligible. Though, I agree it would be better to ship at least one adapter and that's why it's proposed in the pull-request description :)